signac/core/jsondict.py

```diff
@@ -194,8 +194,41 @@ def buffer_reads_writes(buffer_size=DEFAULT_BUFFER_SIZE, force_write=False):
 
 
 class JSONDict(SyncedAttrDict):
-
-    def __init__(self, parent=None, filename=None, write_concern=False):
+    """A dict-like mapping interface to a persistent JSON-file.
+
+    The JSONDict is a :class:`~collections.abc.MutableMapping` and therefore
+    behaves similar to a :class:`dict`, but all data is stored persistently in
+    the associated JSON-file on disk.
+
+    .. code-block:: python
+
+            doc = JSONDict('data.json', write_concern=True)
+            doc['foo'] = "bar"
+            assert doc.foo == doc['foo'] == "bar"
+            assert 'foo' in doc
+            del doc['foo']
+
+    This class allows access to keys both with the slicing and attributes.
+    That means ``doc.foo`` and ``doc['foo']`` are equivalent.
+    Nested keys can also be accessed like that:
+
+    .. code-block:: python
+
+            >>> doc['foo'] = dict(bar=True)
+            >>> doc
+            {'foo': {'bar': True}}
+            >>> doc.foo.bar = False
+            {'foo': {'bar': False}}
+
+    :param filename:
+        The filename of the associated JSON-file on disk.
```

Comment:
JSON file (no hyphen)

Reply:
We use the formulation `HDF5-file` everywhere else, change that, too?

Reply:
Yes please!